bin/main.ml

```diff
@@ -96,7 +103,11 @@ let main () =
     | entered_length -> rec_get_valid_length (int_of_string entered_length)
   in
   let notes = create_notes piano scale in
-  let seed = generate_seed length (List.length notes) in
+  let mid_seg_len =
```

Comment:
What happens if the input is shorter than the beginning + ending length? Would mid_seg_len be negative, or should we constrict the input to > beginning + ending length?

Reply:
note: we decided to make length of melody at least 10 notes